sklearn/multiclass.py

```diff
@@ -1039,6 +1039,16 @@ def predict(self, X):
             Predicted multi-class targets.
         """
         check_is_fitted(self)
-        Y = np.array([_predict_binary(e, X) for e in self.estimators_]).T
-        pred = euclidean_distances(Y, self.code_book_).argmin(axis=1)
+        # ArgKmin only accept C-contiguous array. The aggregated predictions need to be
+        # transposed. We therefore create a F-contiguous array to avoid a copy and have
+        # a C-contiguous array after the transpose operation.
+        Y = np.array([_predict_binary(e, X) for e in self.estimators_], order="F").T
```

Comment:
May you run a benchmark to compare peak memory and runtime performance between `main` and this PR?

Reply:
I don't see any regression in terms of memory usage.

This PR:

```
peak memory: 364.31 MiB, increment: 103.70 MiB
```

In 1.2.0:

```
peak memory: 387.33 MiB, increment: 115.27 MiB
```

<details>

```python
from sklearn.datasets import make_classification
from sklearn.multiclass import OutputCodeClassifier
from sklearn.tree import DecisionTreeClassifier

X, y = make_classification(
    n_samples=100_000,
    n_features=100,
    n_informative=20,
    n_classes=20,
    n_clusters_per_class=1,
    random_state=0,
)
ecoc = OutputCodeClassifier(
    estimator=DecisionTreeClassifier(max_depth=2, random_state=0),
    code_size=3,
    n_jobs=-1,
    random_state=0,
)
ecoc.fit(X, y)

# %%
%load_ext memory_profiler

# %%
%memit ecoc.predict(X)
```

</details>